api/src/infrastructure/docker.rs

```diff
@@ -490,6 +531,73 @@ impl DockerInfrastructure {
         Ok(())
     }
 
+    async fn get_existing_volumes(app_name: &String) -> Result<Vec<VolumeInfo>, ShipLiftError> {
+        let docker = Docker::new();
+        docker.volumes().list().await.map(|volume_infos| {
+            volume_infos
+                .into_iter()
+                .filter(|s| {
+                    s.labels
+                        .as_ref()
+                        .map(|label| label.get(APP_NAME_LABEL) == Some(app_name))
+                        .unwrap_or(false)
+                })
+                .collect::<Vec<VolumeInfo>>()
+        })
+    }
+
+    async fn create_docker_volume(
+        app_name: &String,
+        config: &ServiceConfig,
+    ) -> Result<String, ShipLiftError> {
+        let docker = Docker::new();
+        let volumes = docker.volumes();
+
+        let mut labels: HashMap<&str, &str> = HashMap::new();
+        labels.insert(APP_NAME_LABEL, app_name);
+        labels.insert(SERVICE_NAME_LABEL, &config.service_name());
+
+        let volume_options = VolumeCreateOptions::builder().labels(&labels).build();
+        volumes
+            .create(&volume_options)
+            .await
+            .map(|volume_info| volume_info.name)
+    }
+
+    async fn create_host_config_binds(
+        app_name: &String,
+        image_info: Option<&ImageInfo>,
+        existing_volume: &Vec<VolumeInfo>,
```

Comment:
Typo and broader context

```suggestion
        existing_volumes: &[VolumeInfo],
```